contracts/ServiceManagerToken.sol

```diff
@@ -249,8 +337,15 @@ contract ServiceManagerToken is GenericManager, OperatorSignedHashes {
         bytes memory data
     ) external returns (address multisig)
     {
+        // TODO Check if multisig was updated or not?
+        // Create or update multisig instance
         multisig = IService(serviceRegistry).deploy(msg.sender, serviceId, multisigImplementation, data);
+
+        // Register or update corresponding 8004 agent Id
+        (uint256 agentId, bool registered) = IIdentityRegistryBridger(identityRegistryBridger).register(serviceId);
+
         emit CreateMultisig(multisig);
+        emit ServiceAgentRegisteredOrUpdated(serviceId, agentId, registered);
```

Comment:
Can we also add olas agent id? Thinking for subgraph that will simplify lot of things